maggy/optimizer/abstractoptimizer.py

```diff
@@ -24,9 +27,26 @@ def __init__(self):
         self.trial_store = None
         self.final_store = None
         self.direction = None
+        self.pruner = None
+
+    def initialize(self, exp_dir):
+        """
+        initialize the optimizer and configure logger.
+
+        :param exp_dir: path of experiment directory
+        :rtype exp_dir: str
+        """
+        # init logger of optimizer
+        self.initialize_logger(exp_dir=exp_dir)
+        # optimizer intitialization routine
+        self._initialize()
+        self._log("Initilized Optimizer {}: \n {}".format(self.name(), self.__dict__))
+        # init logger of pruner
+        if self.pruner:
+            self.pruner.initialize_logger(exp_dir=exp_dir)
 
     @abstractmethod
-    def initialize(self):
+    def _initialize(self):
```

Comment:
```suggestion
    def initialize(self):
```
keep as before and change the new one